From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Tue, 29 May 2018 01:51:00 +0200
Subject: authPrompt: Unset preemptiveAnswer on reset

When we get a reset signal the preemptiveAnswer should be also unset or it will
be used next time the user authPrompt will be activated, even without any further
user interaction.

Fixes #311

Bug-GNOME: https://gitlab.gnome.org/GNOME/gnome-shell/issues/311
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/1777956
Forwarded: yes, https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/116
---
 js/gdm/authPrompt.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/gdm/authPrompt.js b/js/gdm/authPrompt.js
index 481cd3a..d9ec7ee 100644
--- a/js/gdm/authPrompt.js
+++ b/js/gdm/authPrompt.js
@@ -439,6 +439,7 @@ var AuthPrompt = new Lang.Class({
         this.verificationStatus = AuthPromptStatus.NOT_VERIFYING;
         this.cancelButton.reactive = true;
         this.nextButton.label = _("Next");
+        this._preemptiveAnswer = null;
 
         if (this._userVerifier)
             this._userVerifier.cancel();
